nh3_testcubes.py

```diff
@@ -1,48 +1,100 @@
 import os
 import sys
 import numpy as np
-import astropy.units as u
 from astropy.io import fits
-from spectral_cube import SpectralCube
 from astropy.utils.console import ProgressBar
 from astropy import log
-import h5py
 log.setLevel('ERROR')
 
-def generate_cubes(nCubes=100, nBorder=1, noise_rms=0.1,
-                   output_dir='random_cubes', fix_vlsr=True,
-                   random_seed=None, remove_low_sep=False, noise_class=False, ml_output=False):
-    """
-    This places nCubes random cubes into the specified output directory
-    """
 
-    if not os.path.isdir(output_dir):
-        os.mkdir(output_dir)
-    xarr11 = spaxis((np.linspace(-500, 499, 1000) * 5.72e-6
-                     + nh3con.freq_dict['oneone'] / 1e9),
-                    unit='GHz',
-                    refX=nh3con.freq_dict['oneone'] / 1e9,
-                    velocity_convention='radio', refX_unit='GHz')
-    xarr22 = spaxis((np.linspace(-500, 499, 1000) * 5.72e-6
-                     + nh3con.freq_dict['twotwo'] / 1e9), unit='GHz',
-                    refX=nh3con.freq_dict['twotwo'] / 1e9,
-                    velocity_convention='radio', refX_unit='GHz')
-
-    # Create holders for ml_output
-    out_arr = []
-    out_y = []
 
-    nDigits = int(np.ceil(np.log10(nCubes)))
+def generate_cubes(nCubes=100, nBorder=1, noise_rms=0.1, output_dir='random_cubes', random_seed=None,
+                   linenames=['oneone', 'twotwo']):
+
+    xarrList = []
+    lineIDList = []
+
+    for linename in linenames:
+        # generate spectral axis for each ammonia lines
+        xarr = spaxis((np.linspace(-500, 499, 1000) * 5.72e-6
+                       + nh3con.freq_dict[linename] / 1e9),
+                      unit='GHz',
+                      refX=nh3con.freq_dict[linename] / 1e9,
+                      velocity_convention='radio', refX_unit='GHz')
+        xarrList.append(xarr)
+
+        # specify the ID fore each line to appear in  saved fits files
+        if linename is 'oneone':
+            lineIDList.append('11')
+        elif linename is 'twotwo':
+            lineIDList.append('22')
+        else:
+            # use line names at it is for lines above (3,3)
+            lineIDList.append(linename)
+
+    # generate random parameters for nCubes
+    nComps, Temp, Width, Voff, logN = generate_parameters(nCubes, random_seed)
+    gradX, gradY = generate_gradients(nCubes, random_seed)
+
+    cubes = []
+
+    for xarr, lineID in zip(xarrList, lineIDList):
+        # generate cubes for each line specified
+        cubeList = []
+        print('----------- generating {0} lines ------------'.format(lineID))
+        for i in ProgressBar(range(nCubes)):
+            cube_i = make_and_write(nCubes, nComps[i], i, nBorder, xarr, Temp[i], Width[i], Voff[i], logN[i], gradX[i], gradY[i]
+                           , noise_rms, lineID, output_dir)
+
+            cubeList.append(cube_i)
+        cubes.append(cubeList)
+
+    return cubes
```

Comment:
The all at once approach seems like it could choke at large numbers of cubes.  Have you tested it?